OfflineService/OfflineServiceViewController.swift

```diff
@@ -0,0 +1,311 @@
+                    print("Unpacked \(unpackedBytes) of \(totalBytes) bytes")
+                } completionHandler: { (numberOfTiles, error) in
+                    do {
+                        if FileManager.default.fileExists(atPath: temporaryPackURL.path) {
+                            try FileManager.default.removeItem(at: temporaryPackURL)
+                        }
+                    } catch {
+                        self.tilesUnpackingLock.unlock()
+                        
+                        print("Failed to remove temporary pack archive. Error: \(error)")
+                        self.presentAlert(OfflineServiceConstants.title, message: error.localizedDescription)
+                    }
+                    
+                    self.tilesUnpackingLock.unlock()
+                    print("Finished unpacking \(numberOfTiles) tiles")
+                }
+            } catch {
+                tilesUnpackingLock.unlock()
+                
+                print("Error occured while unpacking navigation tiles: \(error)")
+                presentAlert(OfflineServiceConstants.title, message: error.localizedDescription)
+            }
+        }
+    }
+    
+    public func onExpired(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function)")
+    }
+    
+    public func onErrored(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function)")
+    }
+    
+    public func onDeleting(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack, callback: @escaping OfflineDataPackAcknowledgeCallback) {
+        print("[OfflineServiceObserver] \(#function)")
+        
+        presentAlert(OfflineServiceConstants.title, message: "Would you like to remove \(metadata.id)?", handler: { _ in
+            callback()
+        })
+    }
+    
+    public func onDeleted(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata) {
+        print("[OfflineServiceObserver] \(#function)")
+        
+        updateOfflineDataRegions(for: domain, metadata: metadata, delete: true)
+    }
+    
+    public func onInitialized() {
+        print("[OfflineServiceObserver] \(#function)")
+    }
+    
+    public func onIdle() {
+        print("[OfflineServiceObserver] \(#function)")
+    }
+    
+    public func onLogMessage(forMessage message: String) {
+        print("[OfflineServiceObserver] \(#function): \(message)")
+        
+        presentAlert(title, message: message)
+    }
+    
+    public var peer: MBXPeerWrapper?
+    
+    // MARK: - UIViewController delegate methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupUI()
+        listAvailableRegions()
```

Comment:
we should show already downloaded regions even if they are not in the list gotten with `offlineService?.listAvailableRegions`, that is needed for cases when the region was downloaded and not presented in the list in response of the `offlineService?.listAvailableRegions`, e.g. it may happen when there is no internet connection

Reply:
Even though 1Tap for Android doesn't seem to support this I think it's a good idea to show already downloaded regions. Added this ability.